eras/conway/impl/src/Cardano/Ledger/Conway/Delegation/Certificates.hs

```diff
@@ -89,48 +130,59 @@ instance NFData (ConwayDelegCert c)
 
 instance NoThunks (ConwayDelegCert c)
 
-data ConwayDCert c
-  = ConwayDCertDeleg !(ConwayDelegCert c)
-  | ConwayDCertPool !(PoolCert c)
-  | ConwayDCertConstitutional !(ConstitutionalDelegCert c)
+data ConwayDCert era
+  = ConwayDCertNoOp !(StakeCredential (EraCrypto era))
```

Comment:
There is really no need to keep around the stake credential, because it is unused. So, my opinion is we have two options:

1. If we keep the credential in the type, then we should call it:
   ```haskell
     = ConwayDCertReg !(StakeCredential (EraCrypto era))
       -- ^ This type is unused and is here as part of deprecation process
   ```
2. Discard the credential completely and call this a noop:
   ```haskell
     = ConwayDCertNoOp
       -- ^ This type is here as part of deprecation process of
       -- of the registration certificate
   ```

Reply:
I think that not having the stake credential breaks the round-trip tests, since there's no way to reconstruct the original CBOR without that information